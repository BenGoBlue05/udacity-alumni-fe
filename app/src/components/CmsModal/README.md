## CmsModal Component
A component that ...

### Example

```js
<CmsModal />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
