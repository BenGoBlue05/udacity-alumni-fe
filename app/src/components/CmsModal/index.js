import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Layer from 'grommet-udacity/components/Layer';
import Heading from 'grommet-udacity/components/Heading';
import Box from 'grommet-udacity/components/Box';
import CheckBox from 'grommet-udacity/components/CheckBox';
import Select from 'grommet-udacity/components/Select';
import Button from 'grommet-udacity/components/Button';
import Form from 'grommet-udacity/components/Form';
import FormField from 'grommet-udacity/components/FormField';
import FormFields from 'grommet-udacity/components/FormFields';
import Footer from 'grommet-udacity/components/Footer';
import { TagEditor } from 'components';

const selectOptions = [
  { label: 'Draft', value: 0 },
  { label: 'Published', value: 1 },
  { label: 'Archived', value: 2 },
];

const CmsModal = ({
  isShowing,
  onClose,
  spotlighted,
  onToggleSpotlight,
  onSetStatus,
  status,
  onSave,
  canSubmit,
  tags,
  onChangeValue,
  featureImage,
  onChangeFeatureImage,
  selectedTags,
}) => (
  <Layer
    hidden={!isShowing}
    onClose={onClose}
    closer
    align="center"
    className={styles.cmsModal}
  >
    <Box pad="medium">
      <Heading align="center">
        Article Settings
      </Heading>
    </Box>
    <Form
      pad="large"
    >
      <FormFields>
        <FormField
          label="Feature Image"
          help="Enter a URL to set the feature image of the article"
        >
          <input
            value={featureImage}
            placeholder="https://c2.staticflickr.com/8/7127/7552248154_978bcb1773.jpg"
            onChange={onChangeFeatureImage}
            type="text"
          />
        </FormField>
        <FormField
          label="Settings"
        >
          <CheckBox
            toggle
            id="spotlight-toggle"
            name="spotlight"
            label="Spotlight"
            onChange={onToggleSpotlight}
            checked={spotlighted}
          />
        </FormField>
        <FormField
          label="Status"
        >
          <Select
            label="Status"
            value={selectOptions[status] ?
              selectOptions[status].label : 'Draft'
            }
            onChange={({ _, option }) => onSetStatus(option)} // eslint-disable-line
            options={selectOptions}
          />
        </FormField>
        {tags && tags.length > 0 &&
          <FormField
            label="Tags"
            className={styles.formField}
          >
            <TagEditor
              onChangeValue={onChangeValue}
              selectedTags={selectedTags}
              tags={tags}
            />
          </FormField>
        }
        </FormFields>
      <Footer align="center" justify="center" pad="medium">
        <Button
          primary
          label="Save and Publish"
          onClick={canSubmit ? onSave : null}
        />
      </Footer>
    </Form>
  </Layer>
);

CmsModal.propTypes = {
  isShowing: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  spotlighted: PropTypes.bool.isRequired,
  onToggleSpotlight: PropTypes.func.isRequired,
  onSetStatus: PropTypes.func.isRequired,
  status: PropTypes.oneOf([0, 1, 2]),
  onSave: PropTypes.func.isRequired,
  canSubmit: PropTypes.bool.isRequired,
  onCreateTag: PropTypes.func.isRequired,
  onChangeFeatureImage: PropTypes.func.isRequired,
  featureImage: PropTypes.string,
};

CmsModal.defaultProps = {
  isShowing: false,
  status: 0,
  spotlighted: false,
};

export default cssModules(CmsModal, styles);
