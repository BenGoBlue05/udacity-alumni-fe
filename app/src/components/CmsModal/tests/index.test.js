import CmsModal from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<CmsModal />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <CmsModal />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
