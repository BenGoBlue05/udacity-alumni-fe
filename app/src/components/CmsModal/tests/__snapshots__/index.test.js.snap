exports[`<CmsModal /> should render with default props 1`] = `
<Layer
  align="center"
  closer={true}
  hidden={true}>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="medium"
    responsive={true}
    tag="div">
    <Heading
      align="center"
      tag="h1">
      Article Settings
    </Heading>
  </Box>
  <Form
    compact={false}
    fill={false}
    pad="large">
    <FormFields>
      <FormField
        label="Settings"
        size="medium"
        strong={false}>
        <CheckBox
          checked={false}
          id="spotlight-toggle"
          label="Spotlight"
          name="spotlight"
          toggle={true} />
      </FormField>
      <FormField
        label="Status"
        size="medium"
        strong={false}>
        <Select
          label="Status"
          onChange={[Function onChange]}
          options={
            Array [
              Object {
                "label": "Draft",
                "value": 0
              },
              Object {
                "label": "Published",
                "value": 1
              },
              Object {
                "label": "Archived",
                "value": 2
              }
            ]
          }
          value="Draft" />
      </FormField>
    </FormFields>
    <Footer
      align="center"
      direction="row"
      justify="center"
      pad="medium"
      responsive={false}>
      <Button
        label="Publish"
        onClick={null}
        type="button" />
    </Footer>
  </Form>
</Layer>
`;
